code/processes/tailsort.q

```diff
@@ -0,0 +1,77 @@
+\d .ts
+
+taildir:hsym `$getenv`KDBTAIL;                                             /-load in taildir env variables
+hdbdir:hsym `$getenv`KDBHDB;                                               /-load in hdb env variables
+savelist:@[value;`savelist;`quote`trade];                                  /-list of tables to save to HDB
+taildbs:key taildir;                                                       /-list of tailDBs that need saved to HDB
+taildirs:();                                                               /-empty list to append tailDB paths to - to be used
+                                                                           / when HDB save is complete to delete tailDB partitions
+\d .
+
+savescompleted:0;                                                          /-variable to count how many tailDBs have been saved to HDB
+
+mergebypart:{[dir;pt;tabname;dest]
+  /-function to merge table partitions from tailDB and save to HDB
+  /-get list of partitions to be merged
+  parts:(key dir) except `access;
+  partdirs:{` sv (x;y;z)}[dir;;tabname] each parts;
+  /-load data from each partition
+  .lg.o[`merge;"reading partition(s) ", (", " sv string[partdirs])];
+  data:get each partdirs;
+  /-if multiple partitions have been read in data will be a list of tabs, if this is the case - join into single tab
+  if[98<>type data;data:raze data];
+  /-upsert data to partition in destination directory
+  dest:` sv .Q.par[dest;pt;tabname],`;
+  .[upsert;
+    (dest;data);
+    {[e] .lg.e[`upserttopartition;"failed to save table to disk : ",e];'e}
+  ];
+  };
+
+addpattr:{[hdbdir;pt;tabname]
+  /-load column to add p attribute on
+  pcol:.ds.loadtablekeycols[][tabname];
+  /-add p attr to on-disk table
+  .lg.o[`attr;"adding p attribute to the ",string[pcol]," col in ",string[tabname]];
+  addattr:{[hdbdir;pt;tabname;pcol]
+    @[.Q.par[hdbdir;pt;tabname];pcol;`p#]
+  };
+  .[addattr;
+    (hdbdir;pt;tabname;pcol);
+    {[e] .lg.e[`attr;"Failed to add attr : ",e];e}
```

Comment:
in this error trap, we don't signal the error, we just return it. the semi-colon on line 42 means that the return here is not used for anything, so as it is currently, there is no point in returning the error string

```suggestion
    {[e] .lg.e[`attr;"Failed to add attr : ",e]}
```

alternatively I suppose we may want to signal it like in `mergebypart` (if we do actually want to signal it there)

```suggestion
    {[e] .lg.e[`attr;"Failed to add attr : ",e];'e}
```

Reply:
yeah, I don't think this error would need to be signalled so I'll remove the returned error